fix(calendar): Fix `Schedule` error when no events

Spaceface16518 opened the pull request:
Since actual track instances (like "General", etc) are computed from the available appointments, the scheduler can't form groups if there are no appointments available because it never expects there to be no instances of a group when a group is specified. In reality, this is a bug in the scheduler component itself, and should be fixed a different way, perhaps by patching React Scheduler itself, but this will fix it for now by not attempting to render groupings when there are no groupings available.

fixes #191

Spaceface16518 committed:
fix(calendar): Only render track groups if there are tracks available

Since actualy track instances (like "General", etc) are computed from the
available appointments, the scheduler can't form groups if there are no
appointments available because it never expects there to be no instances of a
group when a group is specified. In reality, this is a bug in the scheduler
component itself, and should be fixed a different way, perhaps by patching React
Scheduler itself, but this will fix it for now by not attempting to render
groupings when there are no groupings available.

fixes #191

geeth24 commented on pages/schedule/index.tsx:
```diff
@@ -272,9 +272,18 @@ export default function Calendar(props: { scheduleCard: ScheduleEvent[] }) {
-                <GroupingPanel />
+                {
+                  // since tracks are computed from entries, only show grouping if there are any tracks
+                  uniqueTracks.size > 0 ? (
```
your code is breaking: https://hackportal-git-fix-191-acmutd.vercel.app/schedule
`Application Error: A client-side exception has occurred`
<img width="1537" alt="Screenshot 2023-02-25 at 11 58 37 PM" src="https://user-images.githubusercontent.com/47117192/221394862-5edd57a5-4674-464a-a05e-e6e6fadac27b.png">

MichaelZhao21 replied:
So I messed around with this for a lil bit on my local machine and uhhhh for some reason replacing the react fragment tags with an array of components fixes the issue (no clue why):

![image](https://user-images.githubusercontent.com/37679458/221396003-e4632ef6-f341-4c02-a8dd-7d283c5fe031.png)

Spaceface16518 replied:
Whoops I'm dumb. Should be fixed by 608bc0638d760aaac2e2f0e8dfa877915ee2dfec. Not sure how this slipped through.

Spaceface16518 replied:
> So I messed around with this for a lil bit on my local machine and uhhhh for some reason replacing the react fragment tags with an array of components fixes the issue (no clue why):
> 
> ![image](https://user-images.githubusercontent.com/37679458/221396003-e4632ef6-f341-4c02-a8dd-7d283c5fe031.png)

Oh I didn't see this. 608bc0638d760aaac2e2f0e8dfa877915ee2dfec did not work after all, so I'll do this instead.

Spaceface16518 replied:
Okay React is still complaining when I do that. The only way I can get it to work is if I render each in separate template braces.

![image](https://user-images.githubusercontent.com/33424649/221396693-8ec9f445-8e68-4ad4-9571-6c3fd5791bd7.png)

It's not horribly ugly but it feels kinda wrong. Does anyone know a better way to render groups of components that have to be rendered in a specific order? I don't know why the nested block syntax isn't working.

Spaceface16518 replied:
The issue is that, for some god-forsaken reason, React Scheduler components rely on other components being rendered before them. I don't know why that's a pattern that exists, but it does, so the cleanest solution I can think of is to render each component in separate braces. This makes me very sad.

abdullah50053 replied:
React Scheduler is and has been a garbage library, we've just got to work around its BS until we finish the refactor and can move to a different library or create something custom.

abdullah50053 committed:
Merge branch 'develop' into fix/191

Spaceface16518 committed:
fix(calendar): Move `GroupingState` out of conditional render

For some reason, `IntegratedGrouping` freaks out if GroupingState is not
rendered, but `GroupingState` doesn't cause any errors since it just contains
the empty groupings and doesn't try to render anything, so we can just render it
unconditionally and gate the buggy components behind the condition.

geeth24 commented:
still breaking dude...

Spaceface16518 committed:
fix(calendar): Render each grouping component in separate template brace

Since React Scheduler components rely on state of previously rendered
components, the only way I can see to solve this in a clean way is to render
each in a separate template brace. Not the clean solution I would've liked, but
it'll work and it's technically/logically correct.

Spaceface16518 commented:
Yeah I'm very dumb but this will fix it for now, with minimal changes.

MichaelZhao21 reviewed (approved):
lgtm
